lizrd/scripts/grid.py

```diff
@@ -6,244 +6,192 @@
-DRY_RUN = False
-CODE_PATH = os.getcwd()
-INTERACTIVE_DEBUG = False
-RUNS_MULTIPLIER = 1
-PUSH_TO_GIT = False
-COPIED_CODE_PATH = ""
 if __name__ == "__main__":
-    runner = get_machine_backend()
-    HF_DATASETS_CACHE = get_cache_path(runner)
-    SINGULARITY_IMAGE = get_sparsity_image(runner)
-    NODELIST = None
-    N_GPUS = 1
-    CPUS_PER_GPU = 8
-    CUDA_VISIBLE_DEVICES = None
+    CLUSTER_NAME = get_machine_backend()
     PROCESS_CALL_FUNCTION = lambda args, env: subprocess.run(
         [str(arg) for arg in args if arg is not None], env=env
     )
 
     try:
         path = sys.argv[1]
     except IndexError:
-        raise ValueError("config path must be specified as the first argument")
+        raise ValueError("No config path specified. Aborting...")
 
     if path.endswith(".yaml"):
-        grid_args = yaml.safe_load(open(sys.argv[1]))
+        config = yaml.safe_load(open(path))
     else:
         raise ValueError("config path point to a .yaml")
-    grid_args["params"]["path_to_config"] = sys.argv[1]
-
-    RUNNER = grid_args.get("runner", RUNNER)
-    PARAMS = grid_args.get("params", PARAMS)
-    TIME = grid_args.get("time", TIME)
-    GRES = grid_args.get("gres", GRES)
-    DRY_RUN = grid_args.get("dry_run", DRY_RUN)
-    SINGULARITY_IMAGE = grid_args.get("singularity_image", SINGULARITY_IMAGE)
-    HF_DATASETS_CACHE = grid_args.get("hf_datasets_cache", HF_DATASETS_CACHE)
-    RUNS_MULTIPLIER = grid_args.get("runs_multiplier", RUNS_MULTIPLIER)
-    INTERACTIVE_DEBUG = grid_args.get("interactive_debug", INTERACTIVE_DEBUG)
-    PUSH_TO_GIT = grid_args.get("push_to_git", PUSH_TO_GIT)
-    NODELIST = grid_args.get("nodelist", NODELIST)
-    N_GPUS = grid_args.get("n_gpus", N_GPUS)
-    CPUS_PER_GPU = grid_args.get("cpus_per_gpu", CPUS_PER_GPU)
-    CUDA_VISIBLE_DEVICES = grid_args.get("cuda_visible", CUDA_VISIBLE_DEVICES)
-
-    if SINGULARITY_IMAGE is None and runner != MachineBackend.LOCAL:
-        raise ValueError("Singularity image is not specified (in JSON or env variable)")
-
-    if NODELIST is not None:
-        NODELIST = "--nodelist=" + NODELIST
-
-    grid = create_grid(PARAMS)
-    grid = multiply_grid(grid, RUNS_MULTIPLIER)
-    no_experiments = len(grid)
-    minutes_per_exp = timestr_to_minutes(TIME)
-
-    if len(grid) > 1 and runner == MachineBackend.LOCAL and not DRY_RUN:
+
+    config["params"]["path_to_config"] = path
+
+    interactive_debug_session = config.get("interactive_debug", False)
+
+    # list of pairs: a dictionary of training_args and a dictionary of setup_args
+    grid = []
+    total_no_experiments = 0
+    total_minutes = 0
+
+    pprint.pprint(config)
+    single_exp_training_args_grid = create_grid(config["params"])
+
+    setup_args = get_setup_args_with_defaults(config, CLUSTER_NAME)
+    single_exp_training_args_grid = multiply_grid(
+        single_exp_training_args_grid, setup_args["runs_multiplier"]
+    )
+    no_experiments = len(single_exp_training_args_grid)
```

Comment:
AFAIK mamy konwencję, że  używamy n_experiments
```suggestion
    n_experiments = len(single_exp_training_args_grid)
```